File: java/DataManager.java
// Java class with various errors

import java.util.ArrayList;
import java.util.List;
// Missing import: java.util.HashMap

public class DataManager {
    private List<String> items;
    
    public DataManager() {
        this.items = new ArrayList<>();
    }
    
    // Missing return type - syntax error
    public addItem(String item) {
        items.add(item);
    }
    
    public String getItem(int index) {
        if (index >= 0 && index < items.size()) {
            return items.get(index);
        }
        return null;
    }
    
    public int getCount() {
        return items.size();
    }
    
    // Type error: incompatible types
    public String concatenateItems() {
        int result = "";  // Should be String
        for (String item : items) {
            result += item + " ";
        }
        return result;
    }
    
    // Missing import for HashMap
    public HashMap<String, Integer> getItemCounts() {
        HashMap<String, Integer> counts = new HashMap<>();
        for (String item : items) {
            counts.put(item, counts.getOrDefault(item, 0) + 1);
        }
        return counts;
    }
    
    // Syntax error: missing semicolon
    public void clearItems() {
        items.clear()
    }
    
    // Logic error: wrong condition
    public boolean isEmpty() {
        return items.size() > 0;  // Should be == 0 or use items.isEmpty()
    }
}

File: php/data_manager.php
<?php
// PHP class with various errors

class DataManager {
    private $items = [];
    
    public function __construct() {
        $this->items = array();
    }
    
    // Syntax error: missing semicolon
    public function addItem($item) {
        array_push($this->items, $item)
    }
    
    public function getCount(): int {
        return count($this->items);
    }
    
    // Type error: wrong return type
    public function getAllItems(): string {
        return $this->items;  // Should return array
    }
    
    // Syntax error: missing closing brace
    public function clear(): void {
        $this->items = array();
    
    
    // Logic error: wrong condition
    public function isEmpty(): bool {
        return count($this->items) > 0;  // Should be === 0
    }
    
    // Syntax error: missing $ for variable
    public function findItem($value): bool {
        foreach ($this->items as item) {  // Missing $ before item
            if ($item === $value) {
                return true;
            }
        }
        return false;
    }
}

// Type error in function
function concatenateStrings(array $strings): int {
    $result = "";
    foreach ($strings as $str) {
        $result .= $str . " ";
    }
    return $result;  // Returns string but expects int
}

// Syntax error: missing opening brace
function processData($data): array 
    $processed = [];
    foreach ($data as $item) {
        $processed[] = strtoupper($item);
    }
    return $processed;
}

?>
